components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/PolicyManagementServiceImpl.java

```diff
@@ -301,4 +303,55 @@ public Response deactivatePolicies(List<Integer> policyIds) {
         }
     }
 
+    @Override
+    @PUT
+    @Produces("application/json")
+    @Path("apply-changes")
+    public Response applyChanges() {
+        try {
+            PolicyManagerService policyManagementService = DeviceMgtAPIUtils.getPolicyManagementService();
+            PolicyAdministratorPoint pap = policyManagementService.getPAP();
+            pap.publishChanges();
+        } catch (PolicyManagementException e) {
+            String msg = "Exception in applying changes.";
+            log.error(msg, e);
+            throw new UnexpectedServerErrorException(
+                    new ErrorResponse.ErrorResponseBuilder().setCode(500l).setMessage(msg).build());
+        }
+        return Response.status(Response.Status.OK).entity("Changes have been successfully updated.").build();
+    }
+
+    @PUT
+    @Path("/priorities")
+    public Response updatePolicyPriorities(List<PriorityUpdatedPolicyWrapper> priorityUpdatedPolicies) {
+        PolicyManagerService policyManagementService = DeviceMgtAPIUtils.getPolicyManagementService();
+        List<Policy> policiesToUpdate = new ArrayList<>(priorityUpdatedPolicies.size());
+        int i;
+        for (i = 0; i < priorityUpdatedPolicies.size(); i++) {
```

Comment:
Thank you for the comment. I have made the suggested change by changing this to a foreach loop. Please find the PR below with the suggested change. 
https://github.com/wso2/carbon-device-mgt/pull/736

P.S. In here if I keep the value of priorityUpdatedPolicies.size() in a variable and then do the comparison, is there any performance advantage of using foreach loop over this c-style loop? Since this is an ArrayList, .get(i) is O(1) operation. In that case (since I'm using an ArrayList), the loop will be O(n) operation. Therefore, does it make a difference whether I use a c-style loop or a foreach here?